test/index.test.js

```diff
@@ -31,46 +31,90 @@ test('Should not expose jaeger api when explicitly set to false', async ({ teard
   is(fastify.hasRequestDecorator('jaeger'), false)
 })
 
-test('Should initialize plugin with default configuration', async ({ teardown, is, has }) => {
-  delete require.cache[require.resolve('jaeger-client')]
-  delete require.cache[require.resolve(require('path').join(__dirname, '../index'))]
-  require('jaeger-client')
-  const initTracerSpy = spy(require.cache[require.resolve('jaeger-client')].exports.initTracer)
-  require.cache[require.resolve('jaeger-client')].exports.initTracer = initTracerSpy
-
-  const fastify = createFastify()
-
-  teardown(async () => {
-    delete require.cache[require.resolve('jaeger-client')]
-    delete require.cache[require.resolve(require('path').join(__dirname, '../index'))]
-    await fastify.close()
-  })
-
-  fastify.get('/', (req, reply) => {
-    reply.code(200).send({ hello: 'world' })
+test('Should initialize plugin with correct configuration', async (child) => {
+  const serviceName = 'test'
+
+  const setupChildTest = (pluginOpts) => {
+    const clear = () => {
+      delete require.cache[require.resolve('jaeger-client')]
+      delete require.cache[require.resolve(require('path').join(__dirname, '../index'))]
+    }
+
+    clear()
+    require('jaeger-client')
+    const initTracerSpy = spy(require.cache[require.resolve('jaeger-client')].exports.initTracer)
+    require.cache[require.resolve('jaeger-client')].exports.initTracer = initTracerSpy
+
+    const fastify = createFastify()
+    fastify.get('/', (req, reply) => {
+      reply.code(200).send({ hello: 'world' })
+    })
+
+    fastify.register(require('../index'), pluginOpts)
+
+    return {
+      fastify,
+      initTracerSpy,
+      teardown: async () => {
+        clear()
+        await fastify.close()
+      }
+    }
+  }
+
+  child.test('should use default config when no configuration is provided', async (ct) => {
+    const { fastify, initTracerSpy, teardown } = setupChildTest({ serviceName })
+
+    ct.teardown(teardown)
+
+    await fastify.ready()
+    const response = await fastify.inject({ method: 'GET', url: '/' })
+
+    ct.is(initTracerSpy.spy.calls.length, 1)
+    ct.has(initTracerSpy.spy.calls[0][0], {
+      serviceName: 'test',
+      sampler: { type: 'const', param: 1 },
+      reporter: { logSpans: false },
+    })
+    ct.has(initTracerSpy.spy.calls[0][1], { logger: fastify.log })
+    ct.is(fastify.hasRequestDecorator('jaeger'), true)
+    ct.is(response.statusCode, 200)
   })
 
-  fastify.register(require('../index'), { serviceName: 'test' })
-
-  await fastify.ready()
-
-  const response = await fastify.inject({ method: 'GET', url: '/' })
-  is(initTracerSpy.spy.calls.length, 1)
-  has(initTracerSpy.spy.calls[0][0], {
-    serviceName: 'test',
-    sampler: { type: 'const', param: 1 },
-    reporter: { logSpans: false },
-    state: undefined
+  child.test('should merge default config with the user provided config', async (ct) => {
```

Comment:
Please deconstruct tap on the async function to use only the assertions required.

Reply:
✅